Small changes for Slurm Accounting configurations

jdeamicis opened the pull request:
### Description of changes
* Change default user for slurmdbd daemon to same user slurmctld runs with.
* Pass cluster name to pcluster_slurm_config_generator.py as parameter to the python script.

### Tests
* Ran unit tests for the introduced changes.
* Manually created cluster with Slurm accounting enabled.

### References
* #1530 

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

jdeamicis committed:
Small changes for Slurm Accounting configurations

Change default user for slurdbd daemon to same user slurmctld runs with

Reimplement fetcher of the cluster name in pcluster_slurm_config_generator.py
using cfnconfig file

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>

jdeamicis commented on attributes/default.rb:
```diff
@@ -134,8 +134,12 @@
 default['cluster']['slurm']['group_id'] = node['cluster']['slurm']['user_id']
-default['cluster']['slurm']['dbduser'] = 'slurmdbd'
-default['cluster']['slurm']['dbduser_id'] = node['cluster']['reserved_base_uid'] + 5
+default['cluster']['slurm']['dbduser'] = 'slurm'
+default['cluster']['slurm']['dbduser_id'] = if node['cluster']['slurm']['dbduser'].eql?(node['cluster']['slurm']['user'])
+                                              default['cluster']['slurm']['user_id']
+                                            else
+                                              node['cluster']['reserved_base_uid'] + 5
+                                            end
```
@demartinofra what do you think, should we leave this in the hope that having two different users becomes functional in the future? Or do we just remove it from the code?

demartinofra replied:
As agreed we can just drop it for now

jdeamicis reviewed (commented):
Left one comment for @demartinofra

jdeamicis committed:
Small changes for Slurm Accounting configurations

Change default user for slurdbd daemon to same user slurmctld runs with

Reimplement fetcher of the cluster name in pcluster_slurm_config_generator.py
using cfnconfig file

Do not test origin of mysql packages for ubuntu ARM images (they do not come
from repositories we install ourselves)

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>

jdeamicis committed:
Pass cluster-name to pcluster_slurm_config_generator.py

Pass cluster-name as parameter to the python script and remove
evaluation of cluster name from cfnconfig file

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>

jdeamicis committed:
Merge branch 'develop' into wip/devel/slurm_accounting_cookbook
